client/signup/steps/about/index.jsx

```diff
@@ -102,12 +101,16 @@ class AboutStep extends Component {
 
 	setPressableStore = ref => ( this.pressableStore = ref );
 
-	onSiteTopicChange = value => {
-		this.setState( { siteTopicValue: value } );
-		this.props.recordTracksEvent( 'calypso_signup_actions_select_site_topic', { value } );
+	onSiteTopicChange = ( { vertical_name, vertical_slug } ) => {
+		this.setState( {
+			siteTopicValue: vertical_name,
+			siteTopicSlug: vertical_slug,
+		} );
+
+		this.props.recordTracksEvent( 'calypso_signup_actions_select_site_topic', { vertical_name } );
```

Comment:
It's a good question. 

My instinct is to say _'we probably should'_, but now I look at it, it seems as though it's not being used correctly, or at least the descriptor doesn't match the action that triggers the action call. `calypso_signup_actions_select_site_topic` implies that a topic has been _selected_, not that we've received some new input from the keyboard or whenever the topic changes, which is the case now.

It's been [this way for a while it appears](https://github.com/Automattic/wp-calypso/blob/master/client/signup/steps/about/index.jsx#L107).

I can add it to `<SiteTopicStep />` in the same location since it's been around for so long, but in my opinion we should probably refactor the relevant search components to distinguish between changing the state by selecting a topic from the search results and by typing.

Reply:
Thanks for the info @michaeldcain !

> Let's leave it that way until we need know we need to track topic change (and not submission).

No worries, I'll revert the addition.

> can we add a prop to both calypso_signup_actions_submit_site_topic events to track whether or not the value is provided by the endpoint or is_user_input_vertical?

Seems small enough to add here. Will do 👍